pkg/v1/providers/tests/clustergen/run_tests.sh

```diff
@@ -115,6 +115,10 @@ TKR_DATA: |-
       offer: test-offer
       version: test-version
       thirdPartyImage: test-third-party-image
+      gallery: test-gallery
```

Comment:
Since we are essentially hacking in locally a TKR_DATA that really would be been server-injected, I would suggest setting it up in a way that is representative of what we can expect in reality. Since we don't particularly care which test case is fed which kind of azure image metadata, something simple to enhance this hack with is to use SharedGallery info for half the testcases's TKR_DATA and use Marketplace info the rest. wdyt?

Reply:
Updated to break out cases for each provider and special case the sharedGallery image.